src/network/mod.rs

```diff
@@ -307,3 +320,23 @@ pub fn bootstrap(config: &Configuration, blockchain: BlockchainR) {
         }
     }
 }
+
+/// Queries the trusted peers for a block identified with the hash.
+/// The calling thread is blocked until the block is retrieved.
+/// This function is called during blockchain initialization
+/// to retrieve the genesis block.
+pub fn fetch_block(config: &Configuration, hash: &HeaderHash) -> Result<Block, ()> {
+    if config.protocol != Protocol::Grpc {
+        unimplemented!()
+    }
+    match first_trusted_peer_address(config) {
```

Comment:
This will do for now. But we need to have concurrent fetch instead in the long run.